From 7638e0500ca0ecdc9b55761a3642e390ebe2782e Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <kernel@esmil.dk>
Date: Sat, 15 May 2021 00:53:37 +0200
Subject: [PATCH 1/5] riscv: dts: jh7100: Add missing I2C3 entry

..and move board specific properties to jh7100-beaglev-starlight.dts

Signed-off-by: Emil Renner Berthing <kernel@esmil.dk>
---
 .../dts/starfive/jh7100-beaglev-starlight.dts | 26 +++++++++++++++
 arch/riscv/boot/dts/starfive/jh7100.dtsi      | 32 ++++++++-----------
 2 files changed, 40 insertions(+), 18 deletions(-)

diff --git a/arch/riscv/boot/dts/starfive/jh7100-beaglev-starlight.dts b/arch/riscv/boot/dts/starfive/jh7100-beaglev-starlight.dts
index 3698e78d6a21..aad81b3c73de 100644
--- a/arch/riscv/boot/dts/starfive/jh7100-beaglev-starlight.dts
+++ b/arch/riscv/boot/dts/starfive/jh7100-beaglev-starlight.dts
@@ -85,6 +85,14 @@ wifi_pwrseq: wifi-pwrseq {
 };
 
 &i2c0 {
+	clock-frequency = <100000>;
+	i2c-sda-hold-time-ns = <300>;
+	i2c-sda-falling-time-ns = <500>;
+	i2c-scl-falling-time-ns = <500>;
+	scl-gpio = <&gpio 62 0>;
+	sda-gpio = <&gpio 61 0>;
+	status = "okay";
+
 	imx219@10 {
 		compatible = "imx219";
 		reg = <0x10>;
@@ -110,7 +118,25 @@ tda998x@70 {
 	};
 };
 
+&i2c1 {
+	clock-frequency = <400000>;
+	i2c-sda-hold-time-ns = <300>;
+	i2c-sda-falling-time-ns = <100>;
+	i2c-scl-falling-time-ns = <100>;
+	scl-gpio = <&gpio 47 0>;
+	sda-gpio = <&gpio 48 0>;
+	status = "okay";
+};
+
 &i2c2 {
+	clock-frequency = <100000>;
+	i2c-sda-hold-time-ns = <300>;
+	i2c-sda-falling-time-ns = <500>;
+	i2c-scl-falling-time-ns = <500>;
+	scl-gpio = <&gpio 60 0>;
+	sda-gpio = <&gpio 59 0>;
+	status = "okay";
+
 	seeed_plane_i2c@45 {
 		compatible = "seeed_panel";
 		reg = <0x45>;
diff --git a/arch/riscv/boot/dts/starfive/jh7100.dtsi b/arch/riscv/boot/dts/starfive/jh7100.dtsi
index af075f1bef00..bacd2df6299a 100644
--- a/arch/riscv/boot/dts/starfive/jh7100.dtsi
+++ b/arch/riscv/boot/dts/starfive/jh7100.dtsi
@@ -276,12 +276,7 @@ i2c0: i2c@118b0000 {
 			reg = <0x0 0x118b0000 0x0 0x10000>;
 			interrupts = <96>;
 			clocks = <&clkgen JH7100_CLK_I2C0>;
-			clock-frequency = <100000>;
-			i2c-sda-hold-time-ns = <300>;
-			i2c-sda-falling-time-ns = <500>;
-			i2c-scl-falling-time-ns = <500>;
-			scl-gpio = <&gpio 62 0>;
-			sda-gpio = <&gpio 61 0>;
+			status = "disabled";
 		};
 
 		i2c1: i2c@118c0000 {
@@ -292,12 +287,7 @@ i2c1: i2c@118c0000 {
 			reg = <0x0 0x118c0000 0x0 0x10000>;
 			interrupts = <97>;
 			clocks = <&clkgen JH7100_CLK_I2C0>;
-			clock-frequency = <400000>;
-			i2c-sda-hold-time-ns = <300>;
-			i2c-sda-falling-time-ns = <100>;
-			i2c-scl-falling-time-ns = <100>;
-			scl-gpio = <&gpio 47 0>;
-			sda-gpio = <&gpio 48 0>;
+			status = "disabled";
 		};
 
 		i2c2: i2c@12450000 {
@@ -308,12 +298,18 @@ i2c2: i2c@12450000 {
 			reg = <0x0 0x12450000 0x0 0x10000>;
 			interrupts = <74>;
 			clocks = <&clkgen JH7100_CLK_I2C2>;
-			clock-frequency = <100000>;
-			i2c-sda-hold-time-ns = <300>;
-			i2c-sda-falling-time-ns = <500>;
-			i2c-scl-falling-time-ns = <500>;
-			scl-gpio = <&gpio 60 0>;
-			sda-gpio = <&gpio 59 0>;
+			status = "disabled";
+		};
+
+		i2c3: i2c@12460000 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			#clock-cells = <0>;
+			compatible = "snps,designware-i2c";
+			reg = <0x0 0x12460000 0x0 0x10000>;
+			interrupts = <75>;
+			clocks = <&clkgen JH7100_CLK_I2C2>;
+			status = "disabled";
 		};
 
 		trng: trng@118d0000 {
-- 
2.31.1

